drag-and-drop.md

```diff
@@ -0,0 +1,126 @@
+
+```
+package drag
+
+type AppDraggable interface {
+	// StartDrag return true when object ready to start drag-and-drop
+	StartDrag() bool
+	// Payload return Payload to attach to drag-and-drop session
+	Payload() Payload
+}
+```
+
+The initial implementations for **drag.Payload** will include **drag.PayloadString**, **drag.PayloadSliceOfString**. Other types can be added as needed and assuming the platform supports such type.
+
+```
+package drag
+
+type Payload interface {}
+
+type PayloadString string
+type PayloadSliceOfString []string
+
+```
+
+To receive the drag-and-drop from outside the Fyne application the **fyne.CanvasObject** must implement the **fyne.DragReceiver** interface.
+
+The **fyne.Draggable** can implement **drag.Paintable** interface to provide an image that can be used to paint a cursor during drag-and-drop operation.
+
+```
+package drag
+
+type Paintable interface {
+	CursorImage() image.Image
+}
+
+```
+
+## Implementation
+
+### Drag-and-Drop within the Fyne application
+
+The drag and drop within the Fyne application is implemented by finding the component under the mouse pointer and supporting **fyne.DragReceiver** interface in each of the windows of application sorted by z-index. Only the first window with object of **fyne.DragReceiver** will get the call with fyne.Draggable.
+
+The macOS implementation was tested to confirm this design.
+
+This require for each platform to provide a window position, based on its order from front to back among all visible application windows.
+
+The mobile implementation will be concern only with visible window.
+
+### Drop from outside into Fyne application
+
+The existing go-gl/glfw already provide facilities to receive the drop from another application.
+
+The proposed implementation will utilize this functionality to deliver the drop directly to the object inplementing **fyne.DragReceiver** when found.
+
+### Drag to outside of the Fyne application (pending external changes)
+
+To implement the outside Drag-and-Drop the go-gl/glfw changes will be require as there is no support available for such operation.
+
+The driver will check if object implement **drag.AppDraggable**. When object is ready to start a drag (StartDrag() call), the payload will be collected from the object and this information passed to the OS drag-and-drop mechaism.
+
+#### Darwin
+
+For Darwin platform the **NSDraggingDestination** interface already implemented and we can tap into the code to receive the URL pasteboard (already tested).
+However, in order to initiate the drag from Fyne application to another application, the change will require in go-gl/glfw to implement **NSDraggingSource** interface methods in **GLFWContentView** (https://github.com/go-gl/glfw/blob/ea3d685f79fb0206999ffc85ca313ebf6f6b4c51/v3.3/glfw/glfw/src/cocoa_window.m#L629).
+
+#### Windows
+
+TBD - help wanted
+
+#### Linux
+
+TBD - help wanted
+
+#### Mobile
+
+No implementation for Mobile platform.
```

Comment:
Why not? iPad and Android apps can support dragging out of one visible window on to another can't they?

Reply:
agree, we should add this.